python-package/lightgbm/basic.py

```diff
@@ -502,14 +502,19 @@ def c_int_array(data):
 
 
 def _get_bad_pandas_dtypes(dtypes):
-    pandas_dtype_mapper = {'int8': 'int', 'int16': 'int', 'int32': 'int',
-                           'int64': 'int', 'uint8': 'int', 'uint16': 'int',
-                           'uint32': 'int', 'uint64': 'int', 'bool': 'int',
-                           'float16': 'float', 'float32': 'float', 'float64': 'float'}
```

Comment:
Thanks a lot for this PR!
Could you please add one more test that checks LightGBM works with all these types (different `[u]int`s)?

Reply:
I've added a test to compare the results using all float64 and using combinations of uints, ints and floats in e1cf6c9cd759a735dcf5bf68f015e951b255c677. Let me know what you think.